src/ops/cast.cc

```diff
@@ -63,26 +64,29 @@ Op *Cast::create_operator_from_layer(
 //   return hash;
 // }
 
+CastParams Cast::get_params() const {
+  CastParams params;
+  params.dtype = this->dtype;
+  return params;
+}
+
 using PCG::Node;
 Node FFModel::get_or_create_cast_node(const ParallelTensor input,
```

Comment:
I think all `get_or_create_[op]_node` methods should be discarded. If you see a case where that can't be done, lmk and I can take a look

Reply:
I actually do think it's possible for all of them. I was initially a bit confused about `config` for the parallel ops, but it looks like in `substitution.cc` we can take advantage of `model.config` in `GraphXfer::create_new_operator` (so the code in the parallel ops was actually redundant).